model/errors.go

```diff
@@ -0,0 +1,324 @@
+	ErrorTypeValidation     = "https://serverlessworkflow.io/spec/1.0.0/errors/validation"
+	ErrorTypeExpression     = "https://serverlessworkflow.io/spec/1.0.0/errors/expression"
+	ErrorTypeAuthentication = "https://serverlessworkflow.io/spec/1.0.0/errors/authentication"
+	ErrorTypeAuthorization  = "https://serverlessworkflow.io/spec/1.0.0/errors/authorization"
+	ErrorTypeTimeout        = "https://serverlessworkflow.io/spec/1.0.0/errors/timeout"
+	ErrorTypeCommunication  = "https://serverlessworkflow.io/spec/1.0.0/errors/communication"
+	ErrorTypeRuntime        = "https://serverlessworkflow.io/spec/1.0.0/errors/runtime"
+)
+
+type Error struct {
+	// A URI reference that identifies the error type.
+	// For cross-compatibility concerns, it is strongly recommended to use Standard Error Types whenever possible.
+	// Runtimes MUST ensure that the property has been set when raising or escalating the error.
+	Type *URITemplateOrRuntimeExpr `json:"type" validate:"required"`
+	// The status code generated by the origin for this occurrence of the error.
+	// For cross-compatibility concerns, it is strongly recommended to use HTTP Status Codes whenever possible.
+	// Runtimes MUST ensure that the property has been set when raising or escalating the error.
+	Status int `json:"status" validate:"required"`
+	// A short, human-readable summary of the error.
+	Title *StringOrRuntimeExpr `json:"title,omitempty"`
+	// A human-readable explanation specific to this occurrence of the error.
+	Detail *StringOrRuntimeExpr `json:"detail,omitempty"`
+	// A JSON Pointer used to reference the component the error originates from.
+	// Runtimes MUST set the property when raising or escalating the error. Otherwise ignore.
+	Instance *JsonPointerOrRuntimeExpression `json:"instance,omitempty" validate:"omitempty"`
+}
+
+type ErrorFilter struct {
+	Type     string `json:"type,omitempty"`
+	Status   int    `json:"status,omitempty"`
+	Instance string `json:"instance,omitempty"`
+	Title    string `json:"title,omitempty"`
+	Details  string `json:"details,omitempty"`
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("[%d] %s: %s (%s). Origin: '%s'", e.Status, e.Title, e.Detail, e.Type, e.Instance)
+}
+
+// WithInstanceRef ensures the error has a valid JSON Pointer reference
+func (e *Error) WithInstanceRef(workflow *Workflow, taskName string) *Error {
+	if e == nil {
+		return nil
+	}
+
+	// Check if the instance is already set
+	if e.Instance.IsValid() {
+		return e
+	}
+
+	// Generate a JSON pointer reference for the task within the workflow
+	instance, pointerErr := GenerateJSONPointer(workflow, taskName)
+	if pointerErr == nil {
+		e.Instance = &JsonPointerOrRuntimeExpression{Value: instance}
+	}
+	// TODO: log the pointer error
+
+	return e
+}
+
+// newError creates a new structured error
+func newError(errType string, status int, title string, detail error, instance string) *Error {
+	if detail != nil {
```

Comment:
shouldn't the parameters be validated?

Reply:
It's an internal function used only in this file as a shortcut to the builders. I don't think it's necessary.